src/audio/dai.c

```diff
@@ -98,17 +101,31 @@ static void dai_dma_cb(void *arg, enum notify_id type, void *data)
 	}
 
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
-		dma_buffer_copy_to(dd->local_buffer, dd->dma_buffer,
-				   dd->process, bytes);
+		ret = dma_buffer_copy_to(dd->local_buffer, dd->dma_buffer,
+					 dd->process, bytes);
 
 		buffer_ptr = dd->local_buffer->stream.r_ptr;
 	} else {
-		dma_buffer_copy_from(dd->dma_buffer, dd->local_buffer,
-				     dd->process, bytes);
+		ret = dma_buffer_copy_from(dd->dma_buffer, dd->local_buffer,
+					   dd->process, bytes);
 
 		buffer_ptr = dd->local_buffer->stream.w_ptr;
 	}
 
+	if (ret < 0) {
```

Comment:
Can we have an inline comment on what we are checking here.

Reply:
Added in https://github.com/thesofproject/sof/pull/3251